server/sockets/sendMessage.js

```diff
@@ -1,56 +1,61 @@
-const {
-  NEW_EVENT_SEND_MESSAGE,
-  NEW_EVENT_SEND_FAILED,
-  NEW_EVENT_RECEIVE_MESSAGE,
-} = require('../../constants.json');
-const { addMessage, getActiveUser } = require('../utils/lib');
+const { NEW_EVENT_SEND_MESSAGE, NEW_EVENT_SEND_FAILED, NEW_EVENT_RECEIVE_MESSAGE } = require("../../constants.json");
+const { addMessage, getActiveUser } = require("../utils/lib");
+
+// Create an object to store message counts per user
+const messageCounts = {};
 
 module.exports = (socket) => {
   socket.on(
     NEW_EVENT_SEND_MESSAGE,
     async ({ senderId, message, time, chatId }, returnMessageToSender) => {
-      // Below line is just a failed message simulator for testing purposes.
-
-      // const rndInt = Math.floor(Math.random() * 6) + 1;
-      // if (rndInt % 2 !== 0) {
-      //   return;
-      // }
-
       const user = getActiveUser({
         socketId: socket.id,
       });
 
       if (!user) {
         socket.emit(NEW_EVENT_SEND_FAILED, {
-          message:
-            'Hmmm. It seems your login session has expired. ' +
-            'Re-login and try again',
+          message: "Hmmm. It seems your login session has expired. " +
+                   "Re-login and try again",
         });
+        return;
+      }
 
+      // Check the message count for the user
+      const userMessageCount = messageCounts[senderId] || 0;
+
+      if (userMessageCount >= 25) {
+        // User has exceeded the message limit
+        socket.emit(NEW_EVENT_SEND_FAILED, {
+          message: "You have exceeded the message limit. Please try again later.",
+        });
```

Comment:
you also need to give the alert to the user in the frontend

Reply:
Okay...👍🏻

Reply:
What kind of alert messages ! normal alert or using react-toastify ?

Reply:
normal alert

Reply:
remember to do a git pull before a git push